### PR TITLE
Add 5 second timeout to Auth dialer

### PR DESCRIPTION
**Pull request by meyskens**
The TCP level dialer had no timeout before. This could cause some issues. This change adds a configurable timeout of default 5 seconds to the TCP dailer.

```release-note
Add a 5 second timeout to the Mutual Auth TCP handshake
```

### CODE REVIEW

File: pkg/auth/mutual_authhandler.go [hunk 1]
```diff
@@ -53,12 +53,15 @@ func newMutualAuthHandler(logger logrus.FieldLogger, lc hive.Lifecycle, cfg Mutu
 }
 
 type MutualAuthConfig struct {
-	MutualAuthListenerPort int `mapstructure:"mesh-auth-mutual-listener-port"`
+	MutualAuthListenerPort   int           `mapstructure:"mesh-auth-mutual-listener-port"`
+	mutualAuthConnectTimeout time.Duration `mapstructure:"mesh-auth-mutual-Connect-timeout"`
```
Comment (mhofstetter) [inline pkg/auth/mutual_authhandler.go]: typo in `mesh-auth-mutual-Connect-timeout`: needs to be changed to a lowercase `c`

```
2 error(s) decoding:\n\n* '' has invalid keys: mesh-auth-mutual-connect-timeout\n* '' has unset fields: mesh-auth-mutual-Connect-timeout.\nHint: field 'FooBar' matches flag 'foo-bar', or use tag `mapstructure:\"flag-name\"` to match field with flag" subsys=daemon
```

Comment (mhofstetter) [inline pkg/auth/mutual_authhandler.go]: And we should capitalize the field itself `mutualAuthConnectTimeout` -> `MutualAuthConnectTimeout`

Comment (mhofstetter) [inline pkg/auth/mutual_authhandler.go]: we should expose this config property via Helm too

**Review by mhofstetter — CHANGES_REQUESTED** [review]
thanks for this enhancement! :hourglass: :tada: 

some requested changes inline.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 2]
```diff
@@ -1132,6 +1132,7 @@ data:
   mesh-auth-mutual-enabled: "true"
   mesh-auth-mutual-listener-port: {{ .Values.authentication.mutual.port | quote }}
   mesh-auth-spire-agent-socket: {{ .Values.authentication.mutual.spire.agentSocketPath | quote }}
+  mesh-auth-mutual-connect-timeout: {{ .Values.authentication.mutual.connectTimeout | quote }}
```
Comment (mhofstetter) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: Check integration test i failing due to non-updated docs.

```
index 8770ad82..a86974e0 100644
--- a/Documentation/helm-values.rst
+++ b/Documentation/helm-values.rst
@@ -48,6 +48,10 @@
      - Interval for garbage collection of auth map entries.
      - string
      - ``"5m0s"``
+   * - authentication.mutual.connectTimeout
+     - Timeout for connecting to the remote node TCP socket
+     - string
+     - ``"5s"``
    * - authentication.mutual.port
      - Port on the agent where mutual authentication handshakes between agents will be performed
      - int
make[1]: *** [Makefile:75: check] Error 1
HINT: to fix this, run 'make -C Documentation update-helm-values'
```

**Review by mhofstetter — CHANGES_REQUESTED** [review]
helm values needs to be reflected in the docs -> `make -C Documentation update-helm-values` :rocket:

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 2 above)
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: nit: Consider using [`validateDuration`](https://github.com/cilium/cilium/blob/e8618e952098c9c05f2249afe80835d7d37291f7/install/kubernetes/cilium/templates/_helpers.tpl#L146-L156) For Golang duration formatted Helm values, e.g.

https://github.com/cilium/cilium/blob/e8618e952098c9c05f2249afe80835d7d37291f7/install/kubernetes/cilium/templates/cilium-configmap.yaml#L312-L316

```suggestion
  mesh-auth-mutual-connect-timeout: {{ include "validateDuration" .Values.authentication.mutual.connectTimeout | quote }}
```

**Review by kaworu — APPROVED** [review]
Helm changes LGTM.

**Review by mhofstetter — APPROVED** [review]

**Comment by mhofstetter** [comment]
@meyskens looks like we need another round of `make -C Documentation update-helm-values`

**Comment by meyskens** [comment]
Seems the merge un-did the cmdref this time. updating...

**Review by joamaki — APPROVED** [review]

**Pull request merged**